backend.native/tests/interop/platform_zlib.kt

```diff
@@ -2,17 +2,17 @@ import kotlinx.cinterop.*
 
 import platform.zlib.*
 
-val source = immutableBinaryBlobOf(0xF3, 0x48, 0xCD, 0xC9, 0xC9, 0x57, 0x04, 0x00).asCPointer()
+val source = immutableBinaryBlobOf(0xF3, 0x48, 0xCD, 0xC9, 0xC9, 0x57, 0x04, 0x00).asCPointer().reinterpret<UByteVar>()
 val golden = immutableBinaryBlobOf(0x48, 0x65, 0x6C, 0x6C, 0x6F, 0x21, 0x00).asCPointer()
 
 fun main(args: Array<String>) = memScoped {
     val buffer = ByteArray(32)
     buffer.usePinned { pinned ->
         val z = alloc<z_stream>().apply {
             next_in = source
-            avail_in = 8
-            next_out = pinned.addressOf(0)
-            avail_out = buffer.size
+            avail_in = 8u
+            next_out = pinned.addressOf(0).reinterpret()
```

Comment:
Why not UByteArray then?

Reply:
Because `buffer` is decoded to `String` below.